### DISCUSSION TITLE
[New Feature] Extra range from higher ground

### DISCUSSION THREAD

**Discussion by Lylaktech** [Vinifera-Developers/Vinifera#1515] [Feature Requests]
### Description:
In TS, units like tick tank, Disc Thrower or Juggernaut gain extra range against targets that are on lower ground than them. In RA2 this got extended to every weapon in the game gaining this range advantage making it more consistent across all weapons in the game and I'd like to see that in TS, as it makes map design more interesting and dynamic.

I believe the way this range advantage works is it gives +0.5 range for each height unit the weapon is above its' target.

### Possible Implementation:
It could be implemented as SusceptibleToHeight=yes/no on the weapon's code so the modder can decide if they want this feature or not.

**Comment by Rampastring** [comment]
There's already `LeptonsPerFireIncrease` in Rules.ini. I don't know whether it's used though.

**Comment by Crimsonum** [comment]
> It could be implemented as SusceptibleToHeight=yes/no on the weapon's code so the modder can decide if they want this feature or not.

That key name might be a bit misleading. I would name it like in YR, `SubjectToElevation`. In YR it's also a projectile key, not a weapon key. YR also has a couple of other related keys that could be useful, `SubjectToCliffs` and `SubjectToWalls` (the latter I would replace with something like `SubjectToObstacles`, so the projectile can be intercepted by not just walls, but buildings, terrain types, etc.).

> There's already `LeptonsPerFireIncrease` in Rules.ini. I don't know whether it's used though.

`LeptonsPerFireIncrease` doesn't do anything. Only arcing projectiles gain increased range per elevation and this is hardcoded.

However, the related key `LeptonsPerSightIncrease` is used. Problem with this key is that it's *not* relative to nearby terrain. This means that units on low flat terrain will have shorter sight than units on high flat terrain.

That said, I'm not sure how exactly the key works. Considering a cell is 256 leptons high, and the value for this key is 2000 in vanilla rules.ini, a unit should gain increased sight range after climbing 2000 / 256 ≈ 8 cells. However, with the default value units will never gain increased sight. Even with a value like 256 it takes several cells before the sight range is increased, as seen below:
![image](https://user-images.githubusercontent.com/17765965/117569666-67bb9e80-b0cf-11eb-88bb-1a1def7b8d13.png)

Very low values like 10 cause the sight range to fluctuate:
![image](https://user-images.githubusercontent.com/17765965/117569858-3f806f80-b0d0-11eb-999d-808be7c2bcfa.png)
